From b934ba70af5b79fb0913d5c6e30643d080667ca5 Mon Sep 17 00:00:00 2001
From: SanLiy <84434845+adk23333@users.noreply.github.com>
Date: Sun, 22 Mar 2026 15:08:38 +0800
Subject: [GKI PATCH 3/4] GKI 6.12: use Android ABI padding for POSIX_MQUEUE

---
 include/linux/sched/user.h | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/include/linux/sched/user.h b/include/linux/sched/user.h
index 79864440dba9e..db919e1fa6b03 100644
--- a/include/linux/sched/user.h
+++ b/include/linux/sched/user.h
@@ -36,8 +36,12 @@ struct user_struct {
 	/* Miscellaneous per-user rate limit */
 	struct ratelimit_state ratelimit;
 
-	ANDROID_KABI_RESERVE(1);
-	ANDROID_KABI_RESERVE(2);
+#if defined(CONFIG_POSIX_MQUEUE)
+	ANDROID_KABI_USE(1, unsigned long mq_bytes);
+#else
+ 	ANDROID_KABI_RESERVE(1);
+#endif
+ 	ANDROID_KABI_RESERVE(2);
 };
 
 extern int uids_sysfs_init(void);